src/Aspire.Hosting/Dashboard/DcpDataSource.cs

```diff
@@ -104,6 +129,160 @@ await foreach (var (eventType, resource) in _kubernetesService.WatchAsync<T>(can
         }
     }
 
+    private async Task ProcessInitialResource(IResource resource)
+    {
+        ResourceSnapshot? snapshot = default;
+
+        // If the resource is a redirect, we want to create a snapshot for the resource it redirects to.
+        if (resource.TryGetLastAnnotation<ConnectionStringRedirectAnnotation>(out var redirectAnnotation))
```

Comment:
We should change the name of this annotation if we are going to use it in this way ... its scope is now much broader than connection strings.